server/routes/data_source_connection/routes.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { IRouter } from 'opensearch-dashboards/server';
+import { DataSourceAttributes } from '../../../../../src/plugins/data_source/common/data_sources';
+import { API } from '../../../common';
+
+export function registerDataSourceConnectionsRoutes(router: IRouter) {
+  router.get(
+    {
+      path: API.DATA_SOURCE.CONNECTIONS,
+      validate: {
+        params: schema.object({}, { unknowns: 'allow' }),
+      },
+    },
+    async (context, request, response) => {
+      const fields = ['id', 'title', 'auth.type'];
+      const resp = await context.core.savedObjects.client.find<DataSourceAttributes>({
+        type: 'data-source',
+        fields,
+        perPage: 10000,
```

Comment:
should this paginate?

Reply:
it could. and probably should once we start adding more to this wrapper